src/firebase/saga.test.ts

```diff
@@ -0,0 +1,39 @@
+import { initializeFirebase, waitForAction } from 'src/firebase/saga'
+import { expectSaga } from 'redux-saga-test-plan'
+import { call } from 'redux-saga/effects'
+import { getAccount } from 'src/web3/saga'
+import { initializeAuth, initializeCloudMessaging } from 'src/firebase/firebase'
+import firebase from '@react-native-firebase/app'
+
+jest.mock('@react-native-firebase/app', () => ({
+  app: () => ({ options: { databaseURL: 'https://test-firebase.valoraapp.com' } }),
+}))
+jest.mock('src/firebase/firebase')
+jest.mock('@celo/utils/lib/async')
+
+describe('firebase saga', () => {
+  it('waitForAction', async () => {
```

Comment:
notably, this test doesn't verify that the saga does *not* proceed if the action is not dispatched. I tested this manually, but was not able to get it in a unit test here. Open to suggestions of how that could be done

Reply:
I think `expectSaga` [has a default timeout](https://github.com/jfairbank/redux-saga-test-plan/blob/master/docs/integration-testing/timeout.md#timeout) of 250ms by default. I wonder if you could _not_ dispatch the action, and assert that the timeout is indeed triggered? Dunno if that's doable.

Reply:
Joe's suggestion is what I thought of. I pulled the code and tried to write a test around it but it wasn't working. The test would just hit jest's 5000 ms timeout limit

Reply:
i had the same issue